### ISSUE TITLE
getAttachments only works for 2+ attachments

### ISSUE THREAD

**Issue by ElBarto221** [dixudx/rtcclient#160]
Trying to fetch the attachments from a workitem with only one attachment results in

`AttributeError: 'str' object has no attribute 'get'`

Root cause seems to be this code line:

https://github.com/dixudx/rtcclient/blob/dbec1cc87281af3aff9846c1f5f62097b9716406/rtcclient/client.py#L1378

If I change this to 

`if isinstance(entries, dict)`

it works.

**Comment by dixudx** [comment]
Please help submit a PR. Thanks for your contribution.

**Comment by dixudx** [comment]
Actually `OrderedDict` is supposed to work.

@ElBarto221 Would you please help take more tests on this change? We need to make sure the new change works.

**Comment by ElBarto221** [comment]
@dixudx Sure, what do you need me to test? Also, I just realized now that I got the getAttachments method to work, how do I get the actual attachment content?

**Comment by ElBarto221** [comment]
Btw, we are using EWM 7.0.1 with form-based authentication.

**Comment by dixudx** [comment]
> Also, I just realized now that I got the getAttachments method to work, how do I get the actual attachment content?

```python
>>> import pprint
# print the field alias
>>> pprint.pprint(myattachment.field_alias, width=1)
```

You will see all the attribute list, where you can find the content.

**Comment by ElBarto221** [comment]
I tried this

`wk.getAttachments()[0]["content"].get('@rdf:resource') `

but it only gives me the URL. How can I then download the actual attachment content? If I try 

`rtcClient.get(attachments[0]["content"].get('@rdf:resource'))`

it results in 

ExpatError                                Traceback (most recent call last)
File ~\Anaconda3\envs\rtc2jira\lib\site-packages\rtcclient\utils.py:32, in token_expire_handler.<locals>.wrapper(*args, **kwargs)
     31 resp = func(*args, **kwargs)
---> 32 xmltodict.parse(resp.content)
     33 return resp

File ~\Anaconda3\envs\rtc2jira\lib\site-packages\xmltodict.py:378, in parse(xml_input, encoding, expat, process_namespaces, namespace_separator, disable_entities, process_comments, **kwargs)
    377 else:
--> 378     parser.Parse(xml_input, True)
    379 return handler.item

ExpatError: not well-formed (invalid token): line 17, column 78

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
Input In [12], in <cell line: 1>()
----> 1 print(rtcClient.get(attachments[0]["content"].get('@rdf:resource')))

File ~\Anaconda3\envs\rtc2jira\lib\site-packages\rtcclient\utils.py:38, in token_expire_handler.<locals>.wrapper(*args, **kwargs)
     35 if "invalid token" in str(excp):
     36     # expires
     37     try:
---> 38         rtc_obj.relogin()
     39     except RTCException:
     40         raise RTCException("Relogin Failed: "
     41                            "Invalid username or password")

File ~\Anaconda3\envs\rtc2jira\lib\site-packages\rtcclient\client.py:111, in RTCClient.relogin(self)
    106 def relogin(self):
    107     """Relogin the RTC Server/Jazz when the token expires
    108 
    109     """
--> 111     self.log.info("Cookie expires. Relogin to get a new cookie.")
    112     self.headers = None
    113     self.headers = self._get_headers()

TypeError: 'tuple' object is not callable

**Comment by dixudx** [comment]
Please refer to [comment in #161](https://github.com/dixudx/rtcclient/issues/161#issuecomment-1147831593).

```
<             #fixme: if isinstance(entries, OrderedDict):
<             if isinstance(entries, dict):
> ---
>             if isinstance(entries, OrderedDict):
```

**Comment by dixudx** [comment]
@ElBarto221 I found this was due to the dependency `xmltodict`.

`xmltodict` 0.13.0 is buggy. xref martinblech/xmltodict#297

Please stick to older 0.12.0 version. `pip install -U xmltodict==0.12.0`.

**Comment by dixudx** [comment]
> Please stick to older 0.12.0 version. pip install -U xmltodict==0.12.0

This has been fixed in #164.

Close it.
